src/mongocxx/collection.cpp

```diff
@@ -74,9 +74,9 @@ mongocxx::stdx::optional<bsoncxx::document::value> find_and_modify(
         collection, bson_filter.bson(), opts, reply.bson(), &error);
 
     if (!r) {
-        auto gle = mongocxx::libmongoc::collection_get_last_error(collection);
-        mongocxx::throw_exception<mongocxx::write_exception>(
-            bsoncxx::helpers::value_from_bson_t(gle), error);
+        if (!reply.view().empty())
+            mongocxx::throw_exception<mongocxx::write_exception>(reply.steal(), error);
+        mongocxx::throw_exception<mongocxx::write_exception>(error);
```

Comment:
The documentation states that it throws write_exception if it fails, so I think this is correct, for now (until we refactor error handling).